node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1339,10 +1339,11 @@ impl BlockSignaturesCollector {
     ) -> bool {
         self.0.as_ref().map_or(false, |sigs| {
             are_signatures_sufficient_for_sync_to_genesis(
-                consensus::check_sufficient_finality_signatures(
+                consensus::check_sufficient_finality_signatures_with_quorum_formula(
```

Comment:
Will we also finish signatures sync early if we have the weaker quorum in the local storage?

Reply:
In case we get the finality signatures from storage we just short-circuit and return early from the `fetch_and_store_finality_signatures_by_block_header` and that will end the process of fetching sigs for a given block header.

It should be ok, as per this comment, since "_Fetcher<BlockSignatures> will only return the result from storage if the signatures are sufficient_"

Reply:
I wonder whether we should somehow always _try_ to get _(1+f)/2_ for a bit, even though we only _fail_ if we can't get _f_?

Reply:
The way `Fetcher` is written it will first try to get the element from storage and if that fails it starts querying peers. What's the definition of _sufficient_ in the case we get them from storage? Is it _ftt_ or _(1 + ftt) / 2_ ?

Reply:
So the behavior is: If we don't have _(1+f)/2_ in storage, we always request signatures at least once, but then we stop as soon as we have _f_?

Reply:
Yes, that was the initial behavior. It is now updated: https://github.com/casper-network/casper-node/pull/3142#discussion_r908262753